users/models.py

```diff
@@ -0,0 +1,69 @@
+from django.db import models
+from django.utils import timezone
+from django.contrib.auth.models import (AbstractBaseUser,
+                                        PermissionsMixin,
+                                        BaseUserManager)
+
+
+class UserManager(BaseUserManager):
+
+    def create_user(self, email, first_name,
+                    last_name, birth_date, password,
+                    **other_fields):
+
+        if not email:
+            raise ValueError('You must provide an email address')
+
+        if not first_name:
+            raise ValueError('You must provide a first name')
+
+        if not last_name:
+            raise ValueError('You must provide a last name')
+
+        if not birth_date:
+            raise ValueError('You must provide a birth date')
+
+        email = self.normalize_email(email)
+        user = self.model(email=email, first_name=first_name,
+                          last_name=last_name, birth_date=birth_date,
+                          **other_fields)
+        user.set_password(password)
+        user.save()
+        return user
+
+    def create_superuser(self, email, first_name,
+                         last_name, birth_date, password,
+                         **other_fields):
+
+        other_fields.setdefault('is_staff', True)
+        other_fields.setdefault('is_superuser', True)
+
+        if other_fields.get('is_staff') is not True:
```

Comment:
I've changed the if statement to `if not other_fields.get('is_staff')`.

But I had to keep `other_fields.setdefault` because the `get` method does not change the vlaue of the key,
it's only a return parameter that the `get` method returns if it doesn't find the key.

source:
https://www.programiz.com/python-programming/methods/dictionary/get

And I have tried it myself:
`dic = {'arg1': True,
       'arg2': True,
       'arg3': False}

t = dic.get('arg4', False)
print(t)

u = dic.get('arg4')
print(u)

y = dic['arg4']
`
You will get an keyError at `y = dic['arg4']`